code/game/gamemodes/cult/talisman.dm

```diff
@@ -220,25 +220,62 @@
 
 //Talisman of Horrors: Breaks the mind of the victim with nightmarish hallucinations
 /obj/item/weapon/paper/talisman/horror
-	cultist_name = "Talisman of Horrors"
-	cultist_desc = "A talisman that will break the mind of the victim with nightmarish hallucinations."
+	cultist_name = "Talisman of Eldritch Nightmares"
+	cultist_desc = "This talisman will seize upon the paranoia of all nearby crew, making them see everyone as a cultist."
 	color = "#ffb366" // light orange
 	invocation = "Lo'Nab Na'Dm!"
-	creation_time = 80
-
-/obj/item/weapon/paper/talisman/horror/afterattack(mob/living/target, mob/living/user)
-	if(iscultist(user) && (get_dist(user, target) < 7))
-		to_chat(user, "<span class='cultitalic'>You disturb [target] with visions of madness!</span>")
-		if(iscarbon(target))
-			var/mob/living/carbon/H = target
-			H.reagents.add_reagent("mindbreaker", 12)
-			if(is_servant_of_ratvar(target))
-				to_chat(target, "<span class='userdanger'>You see a brief but horrible vision of Ratvar, rusted and scrapped, being torn apart.</span>")
-				target.emote("scream")
-				target.confused = max(0, target.confused + 3)
-				target.flash_act()
-		qdel(src)
-
+	creation_time = 60
+	var/duration = 1200
+	var/list/targets = list()
+	var/list/cultists = list()
+
+/obj/item/weapon/paper/talisman/horror/invoke(mob/living/user, successfuluse = 1)
+	var/image/A
+	var/datum/atom_hud/AH = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	for(var/datum/mind/B in SSticker.mode.cult)
+		world << "Targeting B for cult list"
+		cultists += B.current.client
+	for(var/mob/living/carbon/human/T in range(14, get_turf(src)))
+		if(!iscultist(T) && T.client)
+			world << "Targeting [T]"
+			targets += T.client
+			AH.remove_hud_from(T)
+			addtimer(CALLBACK(GLOBAL_PROC, .proc/hudFix, T), duration)
+			A = image('icons/effects/effects.dmi',T,"bloodsparkles", ABOVE_MOB_LAYER)
+			A.override = 0
+			send_visions(A, cultists, duration)
+	for(var/mob/living/L in GLOB.living_mob_list)
+		if(L.z != user.z) // Does this actually helps performance?
+			continue
+		if(ishuman(L))
+			A = image('icons/mob/mob.dmi',L,"cultist", ABOVE_MOB_LAYER)
+			A.override = 1
+			send_visions(A, targets, duration)
+		else
+			var/construct = pick("floating","artificer","behemoth")
+			A = image('icons/mob/mob.dmi',L,construct, ABOVE_MOB_LAYER)
+			A.override = 1
+			send_visions(A, targets, duration)
+	qdel(src)
+
+/proc/remove_visions(image/I, list/show_to)
+	for(var/client/C in show_to)
+		C.images -= I
+
+/proc/send_visions(image/I, list/show_to, duration)
```

Comment:
We do, but I had to make my own so that I could exclude the targets from their own vision (it feels a lot more convincing, especially if youre alone when youre first hit). 

I didn't feel it was appropriate to add a special check or argument to something that gets called tens of thousands a time every round just to accommodate this item.